### DISCUSSION TITLE
Transaction Scenario vs Entry Scenario when running Log Entry action in flow

### DISCUSSION THREAD

**Discussion by sjames-hms** [jongpie/NebulaLogger#715] [Q&A]
Hi folks, I'm working on building out our logger scenarios to utilize the scenario rules functionality but I'm curious if there are any additional context behind how the transaction scenario is set when using the Log Entry action in flow. When using the Log Entry action, I can set the scenario. When I trigger the flow, the entry scenario on the Log Entry is always correct however the transaction scenario on the Log is sometimes different.

**Comment by jongpie** [comment]
Hi @sjames-hms - the 2 scenario fields have slightly different behavior, so it's expected that they will sometimes be different:
- Entry scenario on `LogEntry__c` - this value should always be whatever value you specify for the entry when you log something (in Flow, Apex, or JavaScript)
- Transaction scenario on `Log__c` - by default, this value is the first scenario value used in a transaction. The intent is that the transaction scenario indicates which of your scenarios was the starting point for a transaction
  - You can also customize it to instead be the last scenario used, via the `LoggerParameter__mdt` record `UseFirstSpecifiedScenario`, though I personally prefer the default behavior

For example, if you have 3 log entries being created in this order:
1. An entry with scenario = "A"
2. An entry with scenario = "B"
3. An entry with scenario = "C"

The default behavior will set "A" as the transaction scenario. Or, if you switch the `LoggerParameter__mdt` record, then it would use "C" as the transaction scenario.

Hope this helps! Let me know if you have any follow-up questions though.

**Reply by sjames-hms** [reply]
@jongpie That makes a lot of sense. I think the weird think is that in my flow, all three Log Entry Actions have the same entry scenario ("A") and I don't have a default scenario set org-wide but the Log has a different scenario ("B"). So I'm not sure what could be setting the first scenario in this case. Maybe there is something I'm missing that's setting the initial scenario. 

It does work correctly when I switch the Logger Parameter for UseFirstSpecifiedScenario to false. But I need to audit/review all of my flows to see if there any issues with useing the last scenario vs the first.

**Reply by jongpie** [reply]
@sjames-hms in that case, it could also be that there's some configuration that's setting the transaction scenario in your org. The field `DefaultScenario__c` on `LoggerSettings__c` (the custom hierarchy settings object) is used that sets the initial value in a transaction. Could you try viewing/querying the `LoggerSettings__c` in your org, and see if any records have the value you're currently seeing stored in `DefaultScenario__c`?

**Reply by sjames-hms** [reply]
@jongpie I doubled checked the LoggerSettings and the default scenario is null. One thing to note is that when I trigger the flow on create, the transaction scenario is correct but when I trigger the flow on update, it's a completely different transaction scenario. So not sure if that has any impact.

**Reply by jongpie** [reply]
@sjames-hms hmmm, in that case, yeah, it doesn't sound like the `LoggerSettings__c` data would have any impact. My other guess is that there's another place in your Flow that's using the mysterious scenario value, combined with possibly some bug in Nebula Logger that's incorrectly using it. If you're able to track down that scenario value, let me know. I'll also see if I can figure out anything else on my side.

**Reply by sjames-hms** [reply]
@jongpie This is happening in more than 1 flow and I'm running into the same issue. Were you able to replicate?

**Reply by jongpie** [reply]
@sjames-hms I unfortunately haven't been able to reproduce this issue, but it still sounds like there's something in your org unintentionally setting the initial scenario. Are you able to find anything in your Flows/Apex that contain the text for the mystery scenario?
